core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/EnableAllocationDecider.java

```diff
@@ -178,7 +178,7 @@ public Decision canRebalance(ShardRouting shardRouting, RoutingAllocation alloca
     }
 
     private static String setting(Allocation allocation, boolean usedIndexSetting) {
-        StringBuilder buf = new StringBuilder("[");
+        StringBuilder buf = new StringBuilder(usedIndexSetting ? "index setting " : "cluster setting ").append("[");
```

Comment:
also, other deciders have same issue with messages (e.g. filter allocation decider). Let's have a separate PR to fix these messages to make clear that they refer to index / cluster settings.